app/models/package.rb

```diff
@@ -516,7 +516,11 @@ def is_singleton_package?
   end
 
   def update_location_quantity(total_quantity, location_id)
-    packages_locations.where(location_id: location_id).first.update_column(:quantity, total_quantity)
+    get_packages_location_with_location_id(location_id).update_quantity(total_quantity)
+  end
+
+  def get_packages_location_with_location_id(location_id)
```

Comment:
No, Currently we are not using it anywhere else. Will remove this method and we can make it separate whenever we will use it multiple times.